MediaBrowser.MediaEncoding/Subtitles/SubtitleEncoder.cs

```diff
@@ -167,23 +167,28 @@ async Task<Stream> ISubtitleEncoder.GetSubtitles(BaseItem item, string mediaSour
 
         private async Task<Stream> GetSubtitleStream(SubtitleInfo fileInfo, CancellationToken cancellationToken)
         {
-            if (fileInfo.IsExternal)
+            if (fileInfo.Protocol == MediaProtocol.Http)
             {
-                var stream = await GetStream(fileInfo.Path, fileInfo.Protocol, cancellationToken).ConfigureAwait(false);
-                await using (stream.ConfigureAwait(false))
+                var result = await DetectCharset(fileInfo.Path, fileInfo.Protocol, cancellationToken).ConfigureAwait(false);
+                var detected = result.Detected;
+
+                if (detected is not null)
                 {
-                    var result = await CharsetDetector.DetectFromStreamAsync(stream, cancellationToken).ConfigureAwait(false);
-                    var detected = result.Detected;
-                    stream.Position = 0;
+                    _logger.LogDebug("charset {CharSet} detected for {Path}", detected.EncodingName, fileInfo.Path);
 
-                    if (detected is not null)
-                    {
-                        _logger.LogDebug("charset {CharSet} detected for {Path}", detected.EncodingName, fileInfo.Path);
+                    using var response = await _httpClientFactory.CreateClient(NamedClient.Default)
+                        .GetAsync(new Uri(fileInfo.Path), HttpCompletionOption.ResponseHeadersRead, cancellationToken)
```

Comment:
Why ResponseHeadersRead?

Reply:
To prevent double buffering (inside httpclient and memorystream)